pkg/bufferdecoder/eventsreader.go

```diff
@@ -150,8 +154,7 @@ func ReadArgFromBuff(ebpfMsgDecoder *EbpfDecoder, params []trace.ArgMeta) (trace
 			return argMeta, nil, fmt.Errorf("error reading byte array size: %v", err)
 		}
 		// error if byte buffer is too big (and not a network event)
-		// TODO: check for network event IDs range isntead of arg name
-		if size > 4096 && argMeta.Name != "payload" {
+		if size > 4096 && id > events.MaxNetID && id < events.MaxCommonID {
```

Comment:
I still think this isn't right.
The condition should probably be:
if size > 4096 && (id < events.NetPacketBase || id > events.MaxNetID) {